aws/provider_test.go

```diff
@@ -22,6 +22,7 @@ var testAccProvidersWithTLS map[string]terraform.ResourceProvider
 var testAccProviderFactories func(providers *[]*schema.Provider) map[string]terraform.ResourceProviderFactory
 var testAccProvider *schema.Provider
 var testAccTemplateProvider *schema.Provider
+var testAccProviderFunc func() *schema.Provider
```

Comment:
Can also be used in other cross-region acceptance tests:
```console
$ grep 'return testAccProvider' *.go
provider_test.go:	testAccProviderFunc = func() *schema.Provider { return testAccProvider }
resource_aws_docdb_cluster_test.go:	return testAccCheckDocDBClusterExistsWithProvider(n, v, func() *schema.Provider { return testAccProvider })
resource_aws_instance_test.go:	return testAccCheckInstanceExistsWithProvider(n, i, func() *schema.Provider { return testAccProvider })
resource_aws_neptune_cluster_test.go:	return testAccCheckAWSNeptuneClusterExistsWithProvider(n, v, func() *schema.Provider { return testAccProvider })
resource_aws_rds_cluster_test.go:	return testAccCheckAWSClusterExistsWithProvider(n, v, func() *schema.Provider { return testAccProvider })
resource_aws_route53_zone_association_test.go:	return testAccCheckRoute53ZoneAssociationExistsWithProvider(n, zone, func() *schema.Provider { return testAccProvider })
resource_aws_route53_zone_test.go:	return testAccCreateRandomRoute53RecordsInZoneIdWithProvider(func() *schema.Provider { return testAccProvider }, zone, recordsCount)
resource_aws_route53_zone_test.go:	return testAccCheckRoute53ZoneExistsWithProvider(n, zone, func() *schema.Provider { return testAccProvider })
resource_aws_s3_bucket_test.go:	return testAccCheckAWSS3BucketExistsWithProvider(n, func() *schema.Provider { return testAccProvider })
```

Reply:
I'll add a Tech Debt issue to tidy these up.

Reply:
https://github.com/terraform-providers/terraform-provider-aws/issues/7702

Reply:
I believe many of the `testAccCheck...ExistsWithProvider()` functions are extraneous or adding unnecessary complexity. Its easier to setup the acceptance testing configurations to only need the "normal" `testAccCheck...Exists()` function with its implicit `testAccProvider` usage.

Reply:
Fixed the merge conflict with the removal of the template provider on merge.